plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/CDK3ServerWizardFragment.java

```diff
@@ -110,20 +110,44 @@ protected String findError() {
 
 		
 		String retString = null;
+		String tmpStr = null;
 		if( homeDir == null || !(new File(homeDir)).exists()) {
-			retString = "The selected file does not exist.";
+			tmpStr = "The selected file does not exist.";
 		} else if( !(new File(homeDir).canExecute())) {
-			retString = "The selected file is not executable.";
-		} else if( minishiftVersionProps != null && !minishiftVersionProps.isValid()) {
-			// We have props but it's missing the expected version key
+			tmpStr = "The selected file is not executable.";
+		}
+		
+		toggleDecorator(homeText, tmpStr);
+		retString = tmpStr;
+		tmpStr = null;
+		if( minishiftVersionProps == null ) {
+			return "Unknown error when checking minishift version: " + homeDir;
+		} else if( !minishiftVersionProps.isValid()) {
 			String err = minishiftVersionProps.getError();
-			String defErr = "Unknown error while checking minishift version";
-			retString = (err == null ? defErr : err);
+			if( err == null ) {
+				err = "Unknown error while checking minishift version";
+			}
+			return err;
+		} else {
+			String versionCompatError = isVersionCompatible(minishiftVersionProps);
+			if( versionCompatError != null )
+				return versionCompatError;
 		}
-		toggleDecorator(homeText, retString);
 		return retString;
 	}
 	
+
+	protected String isVersionCompatible(MinishiftVersions versions) {
+		String cdkVers = versions.getCDKVersion();
+		if( cdkVers == null ) {
+			return "Cannot determine CDK version.";
+		}
+		if( cdkVers.startsWith("3.0.") || cdkVers.startsWith("3.1")) {
```

Comment:
i see this check (negated or not in several places), worth extracting and reusing?